app/models/principal.rb

```diff
@@ -71,8 +71,30 @@ def full_name
     "#{first_name} #{last_name}"
   end
 
+  def next_onboarding_action
+    registered_firms = main_firm_with_trading_names.registered
+
+    if registered_firms.empty?
+      :complete_a_firm
+    elsif any_remote_firms?(registered_firms)
+      :onboarded
```

Comment:
Yes, I like this. This short-circuit makes this much easier to understand what's really happening. :100: